Mathlib/Probability/Distributions/Exponential.lean

```diff
@@ -0,0 +1,281 @@
+  unfold exponentialPdfReal
+  refine Measurable.ite ?hp ((measurable_id'.const_mul r).neg.exp.const_mul r) ?hg
+  · refine MeasurableSet.of_compl ?hp.h
+    apply IsOpen.measurableSet; rw [compl_setOf_le]; exact isOpen_gt' 0
+  · exact measurable_const
+
+/-- The exponential Pdf is strongly measurable -/
+lemma stronglyMeasurable_exponentialPdfReal (r : ℝ) :
+    StronglyMeasurable (E r) :=
+  (measurable_exponentialPdfReal r).stronglyMeasurable
+
+/-- the exponential Pdf is positive for all positive reals-/
+lemma exponentialPdfReal_pos {x r : ℝ} {hr : 0 < r} (hx : 0 < x) :
+    0 < E r x := by
+  simp only [exponentialPdfReal, if_pos hx.le]
+  positivity
+
+/-- The exponential Pdf is nonnegative-/
+lemma exponentialPdfReal_nonneg {r : ℝ} (hr : 0 < r) :
+    ∀ x : ℝ, 0 ≤ E r x := by
+  unfold exponentialPdfReal
+  intro x
+  split_ifs <;> positivity
+
+/-- A negative exponential function is integrable on Intervals in R≥0 -/
+lemma exp_neg_integrableOn_Ioc {b x : ℝ} (hb : 0 < b) :
+    IntegrableOn (fun x ↦ rexp (-(b * x))) (Ioc 0 x) := by
+  simp only [neg_mul_eq_neg_mul]
+  exact (exp_neg_integrableOn_Ioi _ hb).mono_set Ioc_subset_Ioi_self
+
+lemma if_eval_pos {r : ℝ} : ∀ᵐ x : ℝ ∂ volume , x < 0 →
+    ENNReal.ofReal (if x ≥ 0 then r * rexp (-(r * x)) else 0) = 0 := by
+  apply ae_of_all
+  intro x hx
+  replace hx : ¬ 0 ≤ x := by exact not_le.mpr hx
+  simp [hx]
+
+lemma if_eval_neg {r : ℝ} : ∀ᵐ x : ℝ ∂ volume, (x ∈ {x|x ≥ 0} →
+    ENNReal.ofReal (if (x ≥ 0) then (r * rexp (-(r * x))) else 0) =
+    ENNReal.ofReal (r * rexp (-(r * x)))) := by
+  apply ae_of_all
+  intro x hx; split_ifs with h; simp only [ge_iff_le] at h
+  · rfl
+  · contrapose h; simp only [ge_iff_le, not_le, not_lt]; exact hx
+
+lemma antiDeriv_tendsto_zero {r : ℝ} (hr : 0 < r) :
+    Tendsto (fun x ↦ -1/r * exp (-(r * x))) atTop (𝓝 0) := by
+  rw [← mul_zero (-1/r)]
+  apply Tendsto.mul
+  exact tendsto_const_nhds
+  apply tendsto_exp_neg_atTop_nhds_0.comp
+  exact (tendsto_const_mul_atTop_of_pos hr).2 tendsto_id
+
+open Measure
+
+lemma lintegral_exponentialPdfReal_eq_one (r : ℝ) (hr : 0 < r) :
+    ∫⁻ x, exponentialPdf r x = 1 := by
+  rw [lintegral_split (exponentialPdf r) 0, ← ENNReal.toReal_eq_one_iff]
+  have leftSide : ∫⁻ x in {x | x < 0}, exponentialPdf r x = 0 := by
+    simp only [exponentialPdf_eq]
+    rw [set_lintegral_congr_fun (isOpen_gt' 0).measurableSet if_eval_pos, lintegral_zero]
+  have rightSide : ∫⁻ x in {x | x ≥ 0}, exponentialPdf r x
+      = ∫⁻ x in {x | x ≥ 0}, ENNReal.ofReal (r * rexp (-(r * x))) := by
+    simp only [exponentialPdf_eq]
+    exact set_lintegral_congr_fun isClosed_Ici.measurableSet if_eval_neg
+  rw [leftSide]; simp only [ge_iff_le, add_zero]
+  rw [rightSide, ENNReal.toReal_eq_one_iff, ←ENNReal.toReal_eq_one_iff]
+  have hf : 0 ≤ᵐ[(restrictₗ {x : ℝ | x ≥ 0}) ℙ] (fun x ↦ r * (rexp (-(r * x)))) := by
+    apply ae_of_all
+    intro a
+    positivity
+  rw [← restrictₗ_apply, ← integral_eq_lintegral_of_nonneg_ae hf]
```

Comment:
`restrictₗ` is not the standard way to talk about restriction of a measure (it's the restriction as a linear map). use `restrict`.
```suggestion
  have hf : 0 ≤ᵐ[volume.restrict {x : ℝ | x ≥ 0}] (fun x ↦ r * (rexp (-(r * x)))) := by
    apply ae_of_all
    intro a
    positivity
  rw [← integral_eq_lintegral_of_nonneg_ae hf]
```
Also, this could all be simplified to `rw [← integral_eq_lintegral_of_nonneg_ae (ae_of_all _ (fun _ ↦ by positivity))]`, without the `have`. But you don't have to follow that suggestion if you prefer the `have` statement.